### PR TITLE
Upgrade to net5

### PR DESCRIPTION
**Pull request by xssic**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
Upgrade framework and packages to net5

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist:**
- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Api/Authorization/Handlers/ContractRoleAuthHandler.cs [hunk 1]
```diff
@@ -18,24 +18,24 @@ public ContractRoleAuthHandler(IProjectOrgResolver orgResolver)
 
         protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, ContractRole requirement, ContractResource resource)
         {
-            var contract = await orgResolver.ResolveContractAsync(resource.Project.ProjectId, resource.Contract);
+            var contract = await orgResolver.ResolveContractAsync(resource.ProjectId, resource.Contract);
             if (contract == null)
             {
-                requirement.SetEvaluation($"Couldn't locate contract in project '{resource.Project.Name}'");
+                requirement.SetEvaluation($"Couldn't locate contract in project '{resource.ProjectId}'");
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Authorization/Handlers/ContractRoleAuthHandler.cs:24]: Might be relevant to have Project.Name ?
Name do not exist in Fusion.Integration.Authorization.

File: src/backend/api/Fusion.Resources.Api/Authorization/Requirements/ContractRole.cs [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
-        public static ContractRole AnyInternalRole = new ContractRole(RoleType.Any, RoleClassification.Internal);
-        public static ContractRole Any = new ContractRole(RoleType.Any);
-
-
-        public enum RoleType { Any, CompanyRep, ContractResponsible }
-        public enum RoleClassification { Internal, External }
-
-    }
-}
+﻿//using Fusion.AspNetCore.FluentAuthorization;
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Authorization/Requirements/ContractRole.cs:1]: Found in Fusion.Integration.Authorization. 
Can probably be removed. Or?

File: src/backend/api/Fusion.Resources.Api/Fusion.Resources.Api.csproj [hunk 3]
```diff
@@ -18,31 +18,31 @@
...
-    <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.9.10" />
-
-    <PackageReference Include="Swashbuckle.AspNetCore.Filters" Version="5.0.0" />
-    <PackageReference Include="swashbuckle.AspNetCore.Swagger" Version="5.0.0" />
-    <PackageReference Include="swashbuckle.AspNetCore.SwaggerGen" Version="5.0.0" />
-    <PackageReference Include="swashbuckle.AspNetCore.SwaggerUi" Version="5.0.0" />
-
-    <PackageReference Include=" Microsoft.Extensions.Diagnostics.HealthChecks.EntityFrameworkCore" Version="3.1.1" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="3.1.1">
+    <PackageReference Include="FluentValidation.AspNetCore" Version="8.6.1" />
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Fusion.Resources.Api.csproj:21]: There is even a version 9, but aligned with Fusion.AspNetCore version.

File: src/backend/api/Fusion.Resources.Api/Notifications/ContractAllocationNotificationHandler.cs [hunk 4]
```diff
@@ -36,10 +36,17 @@ public async Task Handle(ContractRoleDelegated notification, CancellationToken c
             if (delegatedRole == null)
                 return;
 
-            await notificationClient.CreateNotificationAsync(n => n
-                    .WithRecipient(delegatedRole.Person.AzureUniqueId)
-                    .WithTitle($"You were delegated {delegatedRole.Type} role in {delegatedRole.Contract.ContractNumber} - {delegatedRole.Project.Name}")
-                    .WithDescriptionMarkdown(NotificationDescription.DelegateAssigned(delegatedRole)));
+            await notificationClient.CreateNotificationForUserAsync(delegatedRole.Person.AzureUniqueId,
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Notifications/ContractAllocationNotificationHandler.cs:39]: Fusion.Integration.Notification 5.1.0 required change.
Commented code can be removed.

File: src/backend/api/Fusion.Resources.Api/Notifications/RequestsNotificationHandler.cs [hunk 5]
```diff
@@ -41,10 +41,14 @@ public async Task Handle(RequestDeclinedByCompany notification, CancellationToke
 
             foreach (var recipient in recipients)
             {
-                await notificationClient.CreateNotificationAsync(n => n
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Notifications/RequestsNotificationHandler.cs:44]: Fusion.Integration.Notification 5.1.0 required change.
Commented code can be removed.

File: src/backend/api/Fusion.Resources.Api/Startup.cs [hunk 6]
```diff
@@ -128,33 +121,6 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             {
                 endpoints.MapControllers();
 
-                // TODO: Remove
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Startup.cs:131]: ChaosMonkey to be replaced with nuget package.

File: src/backend/tests/Fusion.Resources.Api.Tests/FusionMocks/NotificationClientMock.cs [hunk 7]
```diff
@@ -1,24 +1,53 @@
...
-using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
+using AdaptiveCards;
+using Fusion.Integration.Notification;
+using Fusion.Integration.Profile;
 
 namespace Fusion.Resources.Api.Tests.FusionMocks
 {
     internal class NotificationClientMock : IFusionNotificationClient
```
Comment (xssic) [inline src/backend/tests/Fusion.Resources.Api.Tests/FusionMocks/NotificationClientMock.cs:10]: Required after bumping packages

File: src/backend/api/Fusion.Resources.Api/Startup.cs [hunk 8]
```diff
@@ -49,7 +45,7 @@ public void ConfigureServices(IServiceCollection services)
                 options.AddFusionAuthorization();
                 options.AddOrgIntegration();
                 options.AddFusionRoles();
-                options.AddFusionNotifications(options => options.OriginatingAppKey = "resources");
+                options.AddFusionNotifications(opts => opts.OriginatingAppKey = "resources");
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Startup.cs:48]: conflicting parameter name adjustment

File: src/backend/api/Fusion.Resources.Api/Fusion.Resources.Api.csproj (hunk 3 above)
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Fusion.Resources.Api.csproj:21]: Would be better to remove the local package ref if you want to align.. Would not be any problem just bumping to latest version here.

File: src/backend/api/Fusion.Resources.Api/Notifications/ContractAllocationNotificationHandler.cs [hunk 9]
```diff
@@ -36,10 +36,17 @@ public async Task Handle(ContractRoleDelegated notification, CancellationToken c
...
+                $"You were delegated {delegatedRole.Type} role in {delegatedRole.Contract.ContractNumber} - {delegatedRole.Project.Name}",
+                builder =>
+                {
+                    builder.AddDescription(NotificationDescription.DelegateAssigned(delegatedRole));
+                });
+
+            /*            await notificationClient.CreateNotificationAsync(n => n
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Notifications/ContractAllocationNotificationHandler.cs:46]: Remove commented code

File: src/backend/api/Fusion.Resources.Api/Notifications/RequestsNotificationHandler.cs (hunk 5 above)
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Notifications/RequestsNotificationHandler.cs:44]: Remove commented code

File: src/backend/api/Fusion.Resources.Api/Startup.cs (hunk 6 above)
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Startup.cs:131]: yepp. Should add this at a later time

**Review by HansDahle — CHANGES_REQUESTED** [review]
Nice, think it's good to go when the notification code has been cleaned up

File: src/backend/api/Fusion.Resources.Api/Fusion.Resources.Api.csproj (hunk 3 above)
Comment (xssic) [reply src/backend/api/Fusion.Resources.Api/Fusion.Resources.Api.csproj]: Maybe incorrect question from my part.
Tried to bump the version to v9, but it failed when running tests.
Only guessing here, but seems like it doesn't play well with other libraries.
Using 8.6.3

**Review by HansDahle — APPROVED** [review]

**Pull request merged**